test/distributed/fsdp/test_fsdp_comm.py

```diff
@@ -0,0 +1,149 @@
+# Owner(s): ["oncall: distributed"]
+
+import sys
+from unittest.mock import patch
+
+import torch
+from torch import distributed as dist
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest, NestedWrappedModule
+from torch.testing._internal.common_utils import (
+    TEST_WITH_DEV_DBG_ASAN,
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+)
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestCommunication(FSDPTest):
+    """Tests ``FullyShardedDataParallel``'s collective communication usage."""
+    @skip_if_lt_x_gpu(2)
+    @parametrize(
+        "nested_model",
+        [False, True],
+    )
+    @parametrize(
+        "use_no_sync",
+        [False, True],
+    )
+    @parametrize(
+        "num_sync_iters",
+        [1, 3],
+    )
+    @parametrize(
+        "num_no_sync_iters",
+        [1, 3],
+    )
+    def test_communication(
+        self,
+        nested_model: bool,
+        use_no_sync: bool,
+        num_sync_iters: int,
+        num_no_sync_iters: int,
+    ):
+        """
+        Tests the ``no_sync()`` context manager by checking the communication
```

Comment:
nit: change the comment to mention that this test is generally for checking the communication cost

Reply:
Great catch. I see why it was confusing. I will update this.